kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/inband/RecordDecryptor.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption.inband;
+
+import java.nio.ByteBuffer;
+import java.util.function.IntFunction;
+
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.record.Record;
+import org.apache.kafka.common.utils.ByteUtils;
+
+import io.kroxylicious.filter.encryption.AadSpec;
+import io.kroxylicious.filter.encryption.CipherCode;
+import io.kroxylicious.filter.encryption.records.RecordTransform;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * A {@link RecordTransform} that decrypts records that were previously encrypted by {@link RecordEncryptor}.
+ */
+public class RecordDecryptor implements RecordTransform {
```

Comment:
I think the javadoc for RecordTransform should state contract that the transform methods that return bytebuffers, the caller owns the bytebuffer and the byte content.